File: LeetCode/1039/golang/lc1039_test.go
package lc1039

import "testing"

func Test_minScoreTriangulation(t *testing.T) {
	testcases := []struct {
		values   []int
		expected int
	}{
		{[]int{1, 2, 3}, 6},
		{[]int{3, 7, 4, 5}, 144},
		{[]int{1, 3, 1, 4, 1, 5}, 13},
	}

	for i, tc := range testcases {
		actual := minScoreTriangulation(tc.values)
		if actual != tc.expected {
			t.Errorf("Testcase %02d (%v) failed: want %d, got %d", i+1, tc.values, tc.expected, actual)
		}
	}
}

File: LeetCode/1039/python/lc1039_test.py
import unittest

from lc1039 import Solution

class Test_minScoreTriangulation(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.minScoreTriangulation([1, 2, 3]), 6)
        self.assertEqual(s.minScoreTriangulation([3, 7, 4, 5]), 144)
        self.assertEqual(s.minScoreTriangulation([1, 3, 1, 4, 1, 5]), 13)

if __name__ == "__main__":
    unittest.main()
